control/nlsys.py

```diff
@@ -1663,87 +1671,190 @@ def ivp_rhs(t, x):
         success=soln.success, message=message)
 
 
-def find_eqpt(sys, x0, u0=None, y0=None, t=0, params=None,
-              iu=None, iy=None, ix=None, idx=None, dx0=None,
-              return_y=False, return_result=False):
-    """Find the equilibrium point for an input/output system.
+class OperatingPoint(object):
+    """A class for representing the operating point of a nonlinear I/O system.
+
+    The ``OperatingPoint`` class stores the operating point of a nonlinear
+    system, consisting of the state and input vectors for the system.  The
+    main use for this class is as the return object for the
+    :func:`find_operating_point` function and as an input to the
+    :func:`linearize` function.
+
+    Attributes
+    ----------
+    states : array
+        State vector at the operating point.
+    inputs : array
+        Input vector at the operating point.
+    outputs : array, optional
+        Output vector at the operating point.
+    result : :class:`scipy.optimize.OptimizeResult`, optional
+        Result from the :func:`scipy.optimize.root` function, if available.
+
+    """
+    def __init__(
+            self, states, inputs=None, outputs=None, result=None,
+            return_outputs=False, return_result=False):
+        self.states = states
+        self.inputs = inputs
+
+        if outputs is None and return_outputs and not return_result:
+            raise SystemError("return_outputs specified by no y0 value")
+        self.outputs = outputs
+        self.return_outputs = return_outputs
+
+        if result is None and return_result:
+            raise SystemError("return_result specified by no result value")
+        self.result = result
+        self.return_result = return_result
+
+    # Implement iter to allow assigning to a tuple
+    def __iter__(self):
+        if self.return_outputs and self.return_result:
+            return iter((self.states, self.inputs, self.outputs, self.result))
+        elif self.return_outputs:
+            return iter((self.states, self.inputs, self.outputs))
+        elif self.return_result:
+            return iter((self.states, self.inputs, self.result))
+        else:
+            return iter((self.states, self.inputs))
+
+    # Implement (thin) getitem to allow access via legacy indexing
+    def __getitem__(self, index):
+        return list(self.__iter__())[index]
+
+    # Implement (thin) len to emulate legacy return value
+    def __len__(self):
+        return len(list(self.__iter__()))
+
+
+def find_operating_point(
+        sys, x0, u0=None, y0=None, t=0, params=None, iu=None, iy=None,
+        ix=None, idx=None, dx0=None, root_method=None, root_kwargs=None,
+        return_outputs=None, return_result=None, **kwargs):
```

Comment:
This needs to be `None` in order for the legacy keyword handling to work: if both `return_y` and `return_outputs` are specified, `_process_legacy_keyword` will raise an exception. Thus with this change, setting the default value of `return_outputs` to False would mean that if you call `find_operating_point` with `return_y`, you get  an exception instead of a warning.